cps2pc

```diff
@@ -0,0 +1,226 @@
+        try:
+            output = subprocess.check_output(['uname', '-s'])
+            return output.strip().lower()
+        except OSError:
+            raise Exception('Unable to detect current platform')
+
+#------------------------------------------------------------------------------
+def get_kind_pc_dictionary(kind, location):
+    kind_dict = {}
+    if kind == 'exe':
+        raise Exception('executable')
+    elif kind in ['dylib', 'static'] and location:
+        # Full path is specified directly without any flag.
+        kind_dict['Libs'] = [' %s' % location]
+    elif kind == 'jar':
+        kind_dict['classpath'] = location
+    elif kind == 'interface':
+        pass
+    else:
+        raise Exception(kind)
+    return kind_dict
+
+#------------------------------------------------------------------------------
+def write_pc(pc_dict, filename, prefix):
+    def list_to_str(l):
+        return ' '.join(l)
+    convert = {'unicode': unicode, 'str': str, 'list': list_to_str}
+    with open(filename, 'w') as f:
+        for k, v in pc_dict.iteritems():
+            f.write('%s: %s\n' % (k, convert[type(v).__name__](v).replace(
+                '@prefix@', prefix)))
+
+#------------------------------------------------------------------------------
+def remove_empty_keys(pc_dict):
+    for k, v in pc_dict.iteritems():
+        if not v:
+            pc_dict.pop(k)
+
+#------------------------------------------------------------------------------
+def optional(extensions, attr):
+    try:
+        return extensions.data[attr]
+    except KeyError:
+        return []
+
+#------------------------------------------------------------------------------
+def write_targets_config(package, output_dir, compiler, prefix):
+    platform = get_platform(package)
+    for component_name, component in package.components.iteritems():
+        # Ignored CPS tags (not supported by pkgconfig):
+        # * Link-Languages
+        # * Configurations
+        pc_dict = OrderedDict()
+        pc_dict['Name'] = component_name
+        pc_dict['Version'] = package.version
+
+        # Handle supplemental schema
+        for element in ['Description', 'Website']:
+            pc_dict[element] = optional(package, element)
+
+        # Add libraries
+        pc_dict['Libs'] = []
+        try:
+            # In theory, we care about both `link_location` and `location`
+            # properties of the component. In practice, Link-Location
+            # supersedes Location for our purposes, and if the CPS did not
+            # specify Link-Location, PyCPS fills `link_location` from Location.
+            pc_dict.update(
+                get_kind_pc_dictionary(component.kind, component.link_location)
+            )
+        except Exception as e:
+            print('Skipping %s: %s kind is not supported by pkgconfig' %
+                  (component_name, e))
+            continue
+
+        pc_dict['Libs'] += ['-l%s' % x for x in component.link_libraries]
```

Comment:
FYI, I'm... not actually sure that `-l` should be added here. The spec isn't clear (partly I think from no one using this part in anger, especially as it's discouraged anyway) if these have implied semantic meaning, or if they're effectively equivalent to Link-Flags, but I suspect the latter.

Reply:
I mean, I think if Link-Libraries is non-empty, the values will *already* be e.g. `-lfoo`. (Or... since you mention it, I guess they might be full paths, also. At any rate, what I'm getting at is that I *think* you just want `pc_dict['Libs'] += component.link_libraries`.)

Reply:
Ok, I understand now.